components/automate-cli/cmd/chef-automate/cmdUtils.go

```diff
@@ -0,0 +1,428 @@
+	}
+}
+
+// IsOpensearchIPs will return the Opensearch Node Ips.
+func isOpensearchIPs(single bool, ip string, infra *AutomteHAInfraDetails) ([]string, error) {
+	if ip != "" {
+		if isValidIP(ip, infra.Outputs.OpensearchPrivateIps.Value) {
+			return []string{ip}, nil
+		} else {
+			return []string{}, errors.New("Please Enter Valid opensearch IP")
+		}
+	} else if single {
+		return []string{infra.Outputs.OpensearchPrivateIps.Value[0]}, nil
+	} else {
+		return infra.Outputs.OpensearchPrivateIps.Value, nil
+	}
+}
+
+// isValidIP will check whether the given ip is in the given remoteservice ips set or not.
+func isValidIP(ip string, ips []string) bool {
+	for _, clusterIP := range ips {
+		if ip == clusterIP {
+			return true
+		}
+	}
+	return false
+}
+
+// printOutput of the remote jobs
+func printOutput(remoteService string, result CmdResult, outputFiles []string, cliWriter *cli.Writer) {
+	var resultFiles []string
+	var err error
+	if result.Error != nil {
+		printErrorMessage(remoteService, result.HostIP, cliWriter, result.Error.Error())
+	} else {
+		printSuccessMessage(remoteService, result.HostIP, cliWriter, result.Output)
+	}
+	if len(result.OutputFiles) != 0 {
+		resultFiles, err = combineOutputFiles(result.HostIP, outputFiles, result.OutputFiles)
+	}
+	if err != nil {
+		cliWriter.Println("Error while combining the output files")
+	}
+	if len(resultFiles) != 0 {
+		for _, file := range resultFiles {
+			cliWriter.Printf("Output file: %s", file+"\n")
+		}
+	}
+}
+
+// printErrorMessage prints the config error message
+func printErrorMessage(remoteService string, hostIP string, cliWriter *cli.Writer, err string) {
+	cliWriter.Fail("Command failed on " + remoteService + " node : " + hostIP + " with error:\n" + err + "\n")
+	cliWriter.BufferWriter().Flush()
+}
+
+// printSuccessMessage prints the config success message
+func printSuccessMessage(remoteService string, hostIP string, cliWriter *cli.Writer, output string) {
+	cliWriter.Printf("Output for Host IP %s : \n%s", hostIP, output+"\n")
+	cliWriter.Success("Command is executed on " + remoteService + " node : " + hostIP + "\n")
+	cliWriter.BufferWriter().Flush()
+}
+
+// combineOutputFiles combines outfile from all node in one file.
+func combineOutputFiles(hostIp string, inputOutputFiles, resultOutputFiles []string) ([]string, error) {
+	var resultFile []string
+	for _, file := range inputOutputFiles {
+		for _, outfile := range resultOutputFiles {
+			if strings.Contains(outfile, file) {
+				result, err := appendChildFileToParentFile(hostIp, file, outfile)
+				if err != nil {
+					return resultFile, err
+				}
+				resultFile = append(resultFile, result)
+
+			}
+		}
+	}
+
+	return resultFile, nil
+}
+
+func appendChildFileToParentFile(hostIp, parent, child string) (string, error) {
+	f, err := os.OpenFile(parent, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+	if err != nil {
+		return "", err
+	}
+	defer f.Close()
+
+	output, err := ioutil.ReadFile(child) // nosemgrep
```

Comment:
Please use `github.com/chef/automate/lib/io/fileutils` package instead of the deprecated `ioutil` package. And avoid skipping semgrep.

Reply:
Discussed